### ISSUE TITLE
Consider not duplicating the crypto API on MatrixClient

### ISSUE THREAD

**Issue by turt2live** [element-hq/element-web#17488]
we really just need a `isCryptoEnabled` flag and the rest can just NPE if people fail to handle `client.crypto` being null....

**Comment by jryans** [comment]
Yes, I've never quite been sure why we duplicate most crypto methods onto the client... 🤔

**Comment by richvdh** [comment]
To update on this: it's in progress as part of my work on Element Web R. We're exposing a public `CryptoApi` interface which is accessible via `client.getCrypto()`, and deprecating the `MatrixClient` wrappers as we add things to `CryptoApi`.

Progress so far includes

 * matrix-org/matrix-js-sdk#3211
 * matrix-org/matrix-js-sdk#3209
 * matrix-org/matrix-js-sdk#3281
 * matrix-org/matrix-js-sdk#3287
 * matrix-org/matrix-js-sdk#3354
 * matrix-org/matrix-js-sdk#3355
 * matrix-org/matrix-js-sdk#3357
 * matrix-org/matrix-js-sdk#3360
